### PR TITLE
(MODULES-6283) Update the PowerShell Manager

### PR DESCRIPTION
**Pull request by RandomNoun7**
This change updates the PowerShell provider in this module to match the
provider code found in the puppetlabs-powershell module. The overall
functionality of the module is not changed.

### CODE REVIEW

File: spec/integration/puppet_x/puppetlabs/iis/powershell_manager_spec.rb [hunk 1]
```diff
@@ -262,7 +241,7 @@ def close_stream(stream, style = :inprocess)
     end
 
     it "should return the exitcode of a script invoked with the call operator &" do
-      fixture_path = File.expand_path(File.dirname(__FILE__) + '../../../../exit-27.ps1')
+      fixture_path = File.expand_path(File.dirname(__FILE__) + '../../../../../exit-27.ps1')
```
Comment (Iristyle) [inline spec/integration/puppet_x/puppetlabs/iis/powershell_manager_spec.rb:287]: Looks like we're missing a `skip` from https://github.com/puppetlabs/puppetlabs-powershell/commit/11aedcc32158df693bf3894d1a72665ff87fe9d0

File: spec/integration/puppet_x/puppetlabs/iis/powershell_manager_spec.rb [hunk 2]
```diff
@@ -174,6 +193,9 @@ def close_stream(stream, style = :inprocess)
       end
 
       it "should create a new PowerShell manager host if the output stream handle is closed" do
+        # currently skipped as it can trigger an internal Ruby thread clean-up race
+        # its unknown why this test fails, but not the identical test against @stderr
+        skip('This test can cause intermittent segfaults in Ruby with w32_reset_event invalid handle')
```
Comment (ThoughtCrhyme) [inline spec/integration/puppet_x/puppetlabs/iis/powershell_manager_spec.rb:198]: Try wrapping it like this:
```
skip "This test can cause intermittent segfaults in Ruby with w32_reset_event invalid handle" do
...
end
```

As per this example https://relishapp.com/rspec/rspec-core/v/3-4/docs/pending-and-skipped-examples/skip-examples

Comment (Iristyle) [reply spec/integration/puppet_x/puppetlabs/iis/powershell_manager_spec.rb:198]: This bit shouldn't be a problem - this is regular rspec and same code works fine in other modules.

I think the comment about "its unknown why this test fails" was an accidental misattribution from @RandomNoun7 based on another beaker-rspec skip that's failing in Jenkins.

File: README.md [hunk 3]
```diff
@@ -163,6 +163,8 @@ iis_virtual_directory { 'vdir':
 
 Allows creation of a new IIS Application and configuration of application parameters.
 
+The iis_application type uses a composite namevar for applicationname and sitename to uniquely identify a declaration. To use this successfully, put both the sitename and the applicationname in the title. Puppet will build the catalog using the composite of the two values, while still using the correct value for the applicationname when creating the IIS application.  It requires a `\` in between the `sitename` and `applicationname`, for example, `iis_application { '#{@site_name}\\#{@app_name}'`.
```
Comment (michaeltlombardi) [inline README.md]: Could the variables be placed in code blocks to improve readability? Especially since they're all lower case.

File: README.md [hunk 4]
```diff
@@ -187,7 +189,7 @@ The name of the application pool for the application.
 
 #### `virtual_directory`
 
-The IIS Virtual Directory to convert to an application on create.
+The IIS Virtual Directory to convert to an application on create. Similar to iis_application, iis_virtual_directory uses composite namevars.
```
Comment (michaeltlombardi) [inline README.md]: Nit: I'd change the wording from `Similar to` to `Like` or `As with` - Similar implies that the implementation is different but doesn't go on to explain how.

File: lib/puppet_x/puppetlabs/iis/powershell_manager.rb [hunk 5]
```diff
@@ -57,87 +89,114 @@ def alive?
...
-        Hash[ props ]
-      ensure
-        CloseHandle(output_ready_event) if output_ready_event
+        out
       end
 
       def exit
         @usable = false
-
+
```
Comment (michaeltlombardi) [inline lib/puppet_x/puppetlabs/iis/powershell_manager.rb]: Nit: Trailing whitespace without any other changes.

**Review by michaeltlombardi — CHANGES_REQUESTED** [review]
Aside from the minor nits, just need to clean up the commits, maybe some fixups and message rewrites. Otherwise, code looks good!

Waiting on adhoc / travis. :)

File: lib/puppet_x/puppetlabs/iis/powershell_manager.rb (hunk 5 above)
Comment (RandomNoun7) [reply lib/puppet_x/puppetlabs/iis/powershell_manager.rb]: Zapped

File: README.md (hunk 4 above)
Comment (RandomNoun7) [reply README.md]: Out of scope for this PR. The commit that caused this to show up in the diff has been rebased out of this PR.

File: README.md (hunk 3 above)
Comment (RandomNoun7) [reply README.md]: Out of scope for this PR. The commit that caused this to show up in the diff has been rebased out of this PR.

**Comment by RandomNoun7** [comment]
Changed as requested.

**Review by michaeltlombardi — APPROVED** [review]

**Pull request merged**